worlds/zork_grand_inquisitor/world.py

```diff
@@ -0,0 +1,210 @@
+
+        region_enum_item: ZorkGrandInquisitorRegions
+        for region_enum_item in region_data.keys():
+            region_mapping[region_enum_item] = Region(region_enum_item.value, self.player, self.multiworld)
+
+        region_locations_mapping: Dict[ZorkGrandInquisitorRegions, Set[ZorkGrandInquisitorLocations]]
+        region_locations_mapping = locations_by_region(include_deathsanity=deathsanity)
+
+        region_enum_item: ZorkGrandInquisitorRegions
+        region: Region
+        for region_enum_item, region in region_mapping.items():
+            regions_locations: Set[ZorkGrandInquisitorLocations] = region_locations_mapping[region_enum_item]
+
+            # Locations
+            location_enum_item: ZorkGrandInquisitorLocations
+            for location_enum_item in regions_locations:
+                data: ZorkGrandInquisitorLocationData = location_data[location_enum_item]
+
+                location: ZorkGrandInquisitorLocation = ZorkGrandInquisitorLocation(
+                    self.player,
+                    location_enum_item.value,
+                    data.archipelago_id,
+                    region_mapping[data.region],
+                )
+
+                location.event = isinstance(location_enum_item, ZorkGrandInquisitorEvents)
+
+                if location.event:
+                    location.place_locked_item(
+                        ZorkGrandInquisitorItem(
+                            data.event_item_name,
+                            ItemClassification.progression,
+                            None,
+                            self.player,
+                        )
+                    )
+
+                location_access_rule: str = location_access_rule_for(location_enum_item, self.player)
+
+                if location_access_rule != "lambda state: True":
+                    location.access_rule = eval(location_access_rule)
+
+                region.locations.append(location)
+
+            # Connections
+            region_exit: ZorkGrandInquisitorRegions
+            for region_exit in region_data[region_enum_item].exits or tuple():
+                entrance_access_rule: str = entrance_access_rule_for(region_enum_item, region_exit, self.player)
+
+                if entrance_access_rule == "lambda state: True":
+                    region.connect(region_mapping[region_exit])
+                else:
+                    region.connect(region_mapping[region_exit], rule=eval(entrance_access_rule))
+
+            self.multiworld.regions.append(region)
+
+    def create_items(self) -> None:
+        quick_port_foozle: bool = self.options.quick_port_foozle.value == 1
+        start_with_hotspot_items: bool = self.options.start_with_hotspot_items.value == 1
+
+        item_pool: List[ZorkGrandInquisitorItem] = list()
+
+        item: ZorkGrandInquisitorItems
+        data: ZorkGrandInquisitorItemData
+        for item, data in item_data.items():
```

Comment:
this could all be changed into a single list comprehension too

Reply:
Not while maintaining the same level of readability. It's a fairly large block.
![image](https://github.com/ArchipelagoMW/Archipelago/assets/35039/476649a0-591c-4233-9d1a-8a7fd40d6730)